target_chains/solana/programs/pyth-solana-receiver/src/lib.rs

```diff
@@ -0,0 +1,260 @@
+            .map_err(|_| ProgramError::InvalidArgument)?;
+        match accumulator_update_data.proof {
+            Proof::WormholeMerkle { vaa, updates: _ } => {
+                let vaa: Vaa<&RawMessage> = serde_wormhole::from_slice(vaa.as_ref()).unwrap();
+                let (header, body): (Header, Body<&RawMessage>) = vaa.into();
+                require_eq!(
+                    <wormhole_sdk::Chain as Into<u16>>::into(body.emitter_chain),
+                    emitter_chain,
+                    ReceiverError::InvalidEmitterChain
+                );
+                let post_vaa_ix_data = PostVAAInstructionData {
+                    version:            header.version,
+                    guardian_set_index: header.guardian_set_index,
+                    timestamp:          body.timestamp,
+                    nonce:              body.nonce,
+                    emitter_chain:      body.emitter_chain.into(),
+                    emitter_address:    body.emitter_address.0,
+                    sequence:           body.sequence,
+                    consistency_level:  body.consistency_level,
+                    payload:            body.payload.to_vec(),
+                };
+                let post_vaa_ix = Instruction {
+                    program_id: ctx.accounts.post_vaa_program.key(),
+                    accounts:   vec![
+                        AccountMeta::new_readonly(ctx.accounts.guardian_set.key(), false),
+                        AccountMeta::new_readonly(ctx.accounts.bridge_config.key(), false),
+                        AccountMeta::new_readonly(ctx.accounts.signature_set.key(), false),
+                        AccountMeta::new(ctx.accounts.vaa.key(), false),
+                        AccountMeta::new(ctx.accounts.payer.key(), true),
+                        AccountMeta::new_readonly(Clock::id(), false),
+                        AccountMeta::new_readonly(Rent::id(), false),
+                        AccountMeta::new_readonly(ctx.accounts.system_program.key(), false),
+                    ],
+                    data:       (POST_VAA, post_vaa_ix_data).try_to_vec()?,
+                };
+                invoke(&post_vaa_ix, ctx.accounts.to_account_infos().as_slice())?;
+            }
+        }
+        Ok(())
+    }
+
+    /// Verify the updates using the posted_vaa account
+    ///  * `vaa_hash` hash post of the post_vaa data to derive the address of the post_vaa account
+    ///  * `emitter_chain` expected emitter_chain from the post_vaa account
+    ///  * `price_updates` Vec of bytes for the updates to verify and post on-chain
+    ///
+    /// TODO:
+    ///    - use a `config` account that can only be modified by governance for checking emitter_chain
+    ///      and other constraints
+    ///
+
+    #[allow(unused_variables)]
+    pub fn post_updates(
+        ctx: Context<PostUpdates>,
+        vaa_hash: [u8; 32], // used for pda seeds
```

Comment:
I don't think you need vaa_hash as a parameters here. We are trusting the posted VAA based on it being owned by wormhole + the discriminator

Reply:
i don't mind removing this but technically any vaa that has been posted by the wormhole program would be owned by wormhole and have the right discriminator. The merkle root prove would fail however in that situation if the wrong vaa was used

Reply:
i removed the `vaa_hash` from the ix parameters but added the signature_set account into the `PostUpdates` context since it also contains the hash & we can verify that it's the correct signature_set since the `vaa` has a field for the signature set pubkey.